src/apim_operations.ts

```diff
@@ -564,6 +564,13 @@ export async function createApimUserIfNotExists(
   if (isSome(maybeExistingApimUser)) {
     return maybeExistingApimUser;
   }
+  // feature flag to temporarily block the creation of new APIM users
+  if (config.lockSelfcareCreateNewApimUser) {
+    logger.debug(
+      "createApimUserIfNotExists|Creating new user locked by config"
+    );
+    return none;
+  }
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are unsure if a problem exists, consider leaving a question.  -->
**issue:** We should prefer the explicitness of business logic over hiding behaviors in the function body. having a function called `create__IfNotExists` makes you think it's an idempotent _getter_ - you either get the entity or an error.
I think it's preferable to have both `createApimUserIfNotExists` and `gerApimUser` and let the caller decide which logic to use. 

Plus: it doesn't require the module `apim_operations` (which it's a collection of agnostic, APIM-related helpers) to know about the application feature flag system.

----

To be clear, in `controllers/idp.ts` below I'd write
```ts
const maybeApimUser = 
  config.lockSelfcareCreateNewApimUser
    ? getApimUser(....)
    : createApimUserIfNotExist(....)
```

 ‏

Reply:
Good advice! Never hide behaviors ;)

Reply:
f822088